lib/Github/HttpClient/Message/ResponseMediator.php

```diff
@@ -65,10 +65,10 @@ public static function getApiLimit(ResponseInterface $response)
         $remainingCalls = (int) $remainingCallsHeader;
 
         if (1 > $remainingCalls) {
-            throw new ApiLimitExceedException((int) $remainingCalls);
+            throw new ApiLimitExceedException($remainingCalls);
         }
 
-        return $remainingCalls;
+        return $remainingCallsHeader;
```

Comment:
Whoops. Maybe in 3.x this function's return type can be changed to int|null.

Reply:
We can prepare it in 2.x already! Maybe something like?

```php
public static function getApiLimit(ResponseInterface $response, $returnInt = false)
    {
        if (false === $returnInt) {
            @trigger_error('Not passing "true" as value for "$returnInt" is deprecated. In v3 this function will return "int|null".', E_USER_DEPRECATED);
        }

        $remainingCallsHeader = self::getHeader($response, 'X-RateLimit-Remaining');

        if (null === $remainingCallsHeader) {
            return null;
        }

        $remainingCalls = (int) $remainingCallsHeader;

        if (1 > $remainingCalls) {
            throw new ApiLimitExceedException($remainingCalls);
        }

        if ($returnInt) {
            return $remainingCalls;
        }

        return $remainingCallsHeader;
    }
```

Reply:
Maybe is overkill. ;)

Reply:
Maybe you are right, let's leave it for now and see how we do it in future releases!